lib/index.js

```diff
@@ -267,36 +267,58 @@ King = new Class({
     /**
      * Generates a valid url for a kong request, cycling through one of the hosts for each call
      * @method module:kingkong#request
-     * @param {String} resource The type of resource to generate a url for ( api, plugin, consumer ) 
+     * @param {String} resource The type of resource to generate a url for ( api, plugin, consumer )
      * @param {String} [id] the id or name of a specific resource
-     * @return {Promise} A promise that will resolve to the result of the request 
+     * @return {Promise} A promise that will resolve to the result of the request
      **/
-    ,request:function _request( action, resource, id, data ){
-        let result = resource_exp.exec( resource )
-          , method = action.toUpperCase()
-          , body   = clone( data )
-          , error
-          , uri
-          ;
+    ,request:function _request( action, resource, id, data, optionalAcceptableStatusCodes ){
+        let requestOptions;
 
-        if(!result){
-            error = new Error('invalid resource type ' + resource);
-            error.name = 'InvalidResourceError';
-            throw error;
+        //  Sometimes we send entire request options object as first parameter in which case
+        //  we don't need to generate the request options.
+        if (typeof action === 'object') {
```

Comment:
I'm not following you here. Per README there is no `request` API and there is no change in externally observable behavior (discounting of course that errors are now reported). README doesn't even mention `sync` API for which I had to dive into code (btw, I think it's a design mistake to even offer `sync` e.g. how do I get the promise from the constructor and check for errors?)

I think that it can be reworked into `_request` maintaining the old `request` and redirecting all internal requests to `_request`. But why even have `request`?